server/migrations/20210925190654-create-user.ts

```diff
@@ -0,0 +1,70 @@
+import { UUIDV4 } from "sequelize";
+
+
+module.exports = {
+  up: async (queryInterface: any, Sequelize: any) => {
+    await queryInterface.createTable("Users", {
+      id: {
+        type: Sequelize.UUID,
+        defaultValue: UUIDV4,
+        allowNull: false,
+        primaryKey: true,
+        unique: true
+      },
+      firstName: {
+        type: Sequelize.STRING,
+        allowNull: false,
+        validate: {
+          isAlpha: true
+        }
+      },
+      lastName: {
+        type: Sequelize.STRING,
+        allowNull: false,
+        validate: {
+          isAlpha: true
+        }
+      },
+      userName: {
+        type: Sequelize.STRING(32),
+        allowNull: false,
+        unique: true
+      },    
+      password: {
+        type: Sequelize.STRING,
+        allowNull: false
+      },   
+      email: {
+        type: Sequelize.STRING(64),
+        allowNull: false,
+        unique: true, /* In the controller, lets force email to lowercase */
+        validate : {
+           isEmail: true, /* Check for valid email format */
+           is: [".*@(mail\.||alum\.||^$)utoronto.ca"],  /* Check for utoronto domain */
+        }
+      },
+      confirmed: {
```

Comment:
This might not be necessary for JWT. The tokens could be stored maybe as a cookie.. JWT tokens can be validated in the backend to see what data it contains. The expiry date is also stored in the token itself.

So if a user makes an API call to create a post, we could send their login token alongside the request. The token can be checked in the back-end to see if it's valid and matches their id, then we proceed to make the post..

Reply:
When you create the user account and send a confirmation email you want to use a token to validate the user. Otherwise anyone can validate the account. Further, when a user issues a password reset request you use the token to verify that the user received and clicked the correct link. The expiry prevents stale password resets. Finally, if user_a changes their password we would clear the token. Thus, user_b wouldn't be able to reuse the password reset link via a replay attack. Now, you could include the user email in the password reset link, but then anyone can just go to example.com?email=foo@bar.com and reset the users password.

This isn't an API token. It's related to account creation and password resets.